mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -55,7 +55,7 @@ class GPU_IndexOp<string mnemonic, list<Trait> traits = []> :
         DeclareOpInterfaceMethods<InferIntRangeInterface>,
         DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>])>,
     Arguments<(ins GPU_DimensionAttr:$dimension)>, Results<(outs Index)> {
-  let assemblyFormat = "$dimension attr-dict";
+  let assemblyFormat = "`` $dimension attr-dict";
```

Comment:
I wonder if this isn't working around a problem instead. Why is the infra inserting an extra leading space here?

(@Mogball for vis)

Reply:
That's likely but I'm not familiar enough with tablegen to fix the world. I'd think that since printing regular SSA values works fine we should be able to also print attributes without these leading spaces. I asked about this here: https://discord.com/channels/636084430946959380/642426447167881246/1230368995291561994

Reply:
@joker-eph would you object to landing this as a workaround and filing an issue for the general problem and fixing that after?

Reply:
ping @joker-eph

Reply:
Yes I would.